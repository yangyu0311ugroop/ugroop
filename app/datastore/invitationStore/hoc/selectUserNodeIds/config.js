import { INVITATION_STORE_SELECTORS } from 'datastore/invitationStore/selectors';

// TODO: Start selecting from a well-maintained id array in template.calculated.userNodeIds

export const CONFIG_1 = ({ outputProp }) => ({
  value: {
    [outputProp]: INVITATION_STORE_SELECTORS.userNodeIds,
  },
});

export const CONFIG_2 = ({ outputProp, userIds }) => ({
  value: {
    [outputProp]: INVITATION_STORE_SELECTORS.filterUserNodesByUserId({
      ids: outputProp,
      userIds,
    }),
  },
});

export const CONFIG_3 = ({ outputProp, nodeIds }) => ({
  value: {
    [outputProp]: INVITATION_STORE_SELECTORS.filterUserNodesByNodeId({
      ids: outputProp,
      nodeIds,
    }),
  },
});

export const CONFIG_4 = ({ outputProp, roles }) => ({
  value: {
    [outputProp]: INVITATION_STORE_SELECTORS.filterUserNodesByRole({
      ids: outputProp,
      roles,
    }),
  },
});
