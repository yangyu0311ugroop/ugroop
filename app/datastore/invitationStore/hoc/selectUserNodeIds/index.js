import { compose } from 'redux';
import resaga from 'resaga';
import _ from 'lodash';
import { CONFIG_1, CONFIG_2, CONFIG_3, CONFIG_4 } from './config';

/**
 * Select all userNode id's connected to one of a set of nodeIds via one of a set of roles.
 *
 * @param userIds Optional array of userIds to filter by or name of prop
 * @param nodeIds Optional array of nodeIds to filter by or name of prop
 * @param outputProp Prop name of selected userNode id's
 */
export default ({
  userIds = null,
  nodeIds = null,
  outputProp = 'userNodeIds',
  roles = null,
} = {}) =>
  compose(
    ..._.compact([
      resaga(CONFIG_1({ outputProp })),
      userIds && resaga(CONFIG_2({ outputProp, userIds })),
      nodeIds && resaga(CONFIG_3({ outputProp, nodeIds })),
      resaga(CONFIG_4({ outputProp, roles })),
    ]),
  );
