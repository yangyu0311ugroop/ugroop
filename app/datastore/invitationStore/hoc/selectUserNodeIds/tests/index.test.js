import selectUserNodeIds from '..';

describe('datastore/invitationStore/hoc/selectUserNodeIds', () => {
  it('still matches snapshot', () => {
    expect(
      selectUserNodeIds({
        userIds: [1],
        nodeIds: [2],
        r: jest.fn(args => args),
      }),
    ).toMatchSnapshot();
  });

  it('still matches snapshot with resaga', () => {
    expect(selectUserNodeIds()).toMatchSnapshot();
  });
});
