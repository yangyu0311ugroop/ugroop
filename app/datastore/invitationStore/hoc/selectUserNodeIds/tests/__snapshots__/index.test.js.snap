// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datastore/invitationStore/hoc/selectUserNodeIds still matches snapshot 1`] = `[Function]`;

exports[`datastore/invitationStore/hoc/selectUserNodeIds still matches snapshot with resaga 1`] = `[Function]`;
